### ISSUE TITLE
Android DataStoreException: ambiguous column name

### ISSUE THREAD

**Issue by filippomenchini** [aws-amplify/amplify-flutter#662]
I'm downloading some data from by backend.
This data has one-to-one relations with two entities.
While I'm trying to download this particular data, Amplify.DataStore throws a DataStoreException caused by an SQLiteException saying that there is an ambiguous column name inside the query.
It happens only on Android, on iOS everything works just fine.

**To Reproduce**
1. Create a model schema like this one:
   
<img width="556" alt="Schermata 2021-06-22 alle 17 29 36" src="https://user-images.githubusercontent.com/33838929/122954298-98a21b00-d37f-11eb-8766-053cf2d6b658.png">

2. Try to download some data from the backend with:
    `final tales = await Amplify.DataStore.query(Page.classType);`
3. Start debugging on Android device.
4. It throws the exception.

**Expected behavior**
DataStore should return the list of data I've requested.

**Platform**
Amplify Flutter current supports iOS and Android.  This issue is reproducible in (check all that apply):
[X] Android
[] iOS

**Smartphone (please complete the following information):**
 - Device: Huawei Nova 5T
 - OS: Android 10

<details>
<summary>Exception</summary>
`Caused by: android.database.sqlite.SQLiteException: ambiguous column name: User.id (Sqlite code 1 SQLITE_ERROR): , while compiling: SELECT `Page`.`id` AS `Page_id`, `Page`.`contentKey` AS `Page_contentKey`, `Page`.`pageTaleId` AS `Page_pageTaleId`, `Page`.`pageUserId` AS `Page_pageUserId`, `Tale`.`id` AS `Tale_id`, `Tale`.`title` AS `Tale_title`, `Tale`.`taleUserId` AS `Tale_taleUserId`, `User`.`id` AS `User_id`, `User`.`avatarKey` AS `User_avatarKey`, `User`.`description` AS `User_description`, `User`.`email` AS `User_email`, `User`.`username` AS `User_username`, `User`.`id` AS `User_id`, `User`.`avatarKey` AS `User_avatarKey`, `User`.`description` AS `User_description`, `User`.`email` AS `User_email`, `User`.`username` AS `User_username` FROM `Page` LEFT JOIN `Tale` ON `Page`.`pageTaleId`=`Tale`.`id` LEFT JOIN `User` ON `Tale`.`taleUserId`=`User`.`id`LEFT JOIN `User` ON `Page`.`pageUserId`=`User`.`id`;, (OS error - 2:No such file or directory)`
</details>

<details>
<summary>Output of <code>flutter doctor -v</code></summary>

  ```[✓] Flutter (Channel stable, 2.2.2, on macOS 11.4 20F71 darwin-arm, locale it-IT)
    • Flutter version 2.2.2 at /Users/filippo/flutter
    • Framework revision d79295af24 (11 giorni fa), 2021-06-11 08:56:01 -0700
    • Engine revision 91c9fc8fe0
    • Dart version 2.13.3

[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
    • Android SDK at /Users/filippo/Library/Android/sdk
    • Platform android-30, build-tools 30.0.3
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.8+10-b944.6916264)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 12.5.1, Build version 12E507
    • CocoaPods version 1.10.1

[✗] Chrome - develop for the web (Cannot find Chrome executable at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome)
    ! Cannot find Chrome. Try setting CHROME_EXECUTABLE to a Chrome executable.

[✓] Android Studio (version 4.2)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.8+10-b944.6916264)

[✓] VS Code (version 1.57.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.23.0

[!] Connected device
    ! No devices available

! Doctor found issues in 2 categories.```
</details>

**Comment by HuiSF** [comment]
Hey @filippomenchini thanks for reporting this issue. 

By any chance did you update the schema with breaking changes after writing some data into the local data storage in Android? (if so, please try `Amplify.DataStore.clear()` to clear outdated tables from local, or uninstall App and reinstall)

**Comment by filippomenchini** [comment]
I tried uninstalling and re-installing the app but It didn't work. I have to try the `Amplify.DataStore.clear()`.
I'll let you know of that works.

Thanks for the help!

**Comment by filippomenchini** [comment]
Tried with `Amplify.DataStore.clear()`, unluckly It didn't work.
But i may have found the issue.
I've got two entities wich are related to each other, and they also have a relation with another entity.
Here is a screenshot of the data model:
![image](https://user-images.githubusercontent.com/33838929/123053847-9afa8880-d404-11eb-89b5-d9702f44f67f.png)
The entity Page and the entity Tale are related togheter, and they each have a relation with the User entity.
I think that these relations might be the reason of the Exception, because Amplify.DataStore has to make 2 left joins with "User" and uses the same name for both.
Is something wrong with my design or is there a bug of some sort in Amplify.DataStore?

**Comment by Jordan-Nelson** [comment]
I believe this is caused by a known issue in amplify-android, which is used by amplify-flutter. Here is the issue in amplify-android - https://github.com/aws-amplify/amplify-android/issues/1268

**Comment by filippomenchini** [comment]
@Jordan-Nelson  Is there any way that I could fix this in amplify-flutter? I've just read your solution but I don't think I can do that edit in amplify-flutter. If you have any idea I'm all ears!

**Comment by gabrielgarciagava** [comment]
The issue https://github.com/aws-amplify/amplify-android/issues/1268 was fixed 7 days ago, and released on Amplify Android 1.31.1. When can we expect it to be propagated to Amplify Flutter?

**Comment by Jordan-Nelson** [comment]
Hello @gabrielgarciagava - I don't have an exact date, but there are a few fixes in the latest version of Amplify-Android that we are working to integrate and get out in a release. I will keep this ticket updated.

**Comment by gabrielgarciagava** [comment]
Hey @Jordan-Nelson , I forked the project and updated the Android dependencies. I am using my own fork for now. Of course I don't know all the implications of my update, so I'm not opening any PR.
But if you want to take a look, this is my commit: https://github.com/aws-amplify/amplify-flutter/compare/main...gabrielgarciagava:main

**Comment by HuiSF** [comment]
Hi all, the fix made in amplify-android has been integrated in amplify-flutter, and the fix is available now with amplify-flutter 0.3.2. Please feel free to follow up if anything didn't work as expected.
